app/src/main/jni/interpreter.c

```diff
@@ -1,5 +1,44 @@
 #include "interpreter.h"
 
+/* Verbosely execute Python code.
+ * Exceptions are print to Android log using LOGI macro.
+ * Returns 0 on success or -1 if an exception was raised.
+ */
+int Verbose_PyRun_SimpleString(const char *code) {
+  // Used for globals and locals
+  PyObject *module = PyImport_AddModule("__main__");
+  PyObject *d = PyDict_New();
+  if (module == NULL)
+    return -1;
```

Comment:
Thanks for the feedback.

This code actually copies most of cpython's [PyRun_SimpleStringFlags](https://github.com/python/cpython/blob/0f21fe6155227d11dc02bd3ef3b061de4ecea445/Python/pythonrun.c#L414-L428) method, but instead of printing errors to stderr, it intercepts them and writes them to the Android log.
Cpython handles `module == NULL` exactly the same way, so I didn't spent any further thoughts on it.

Also, the method was moved to [pyhelper.c](https://github.com/aaaaalbert/sensibility-testbed/blob/native-sensors-jni/app/src/main/jni/pyhelper.c) in a later commit, where I added some additional comments. But I do agree that this kind of comments should be added when I frist write the code.